File: RaveX-ravex146/src/main/cpp/addon/AddonLogger.cpp
#include "include/AddonLogger.hpp"
#include <iostream>

namespace ravex {
namespace addon {

void AddonLogger::info(const std::string& prefix, const std::string& msg) {
    std::cout << "[RaveX-Addon] [" << prefix << "] [INFO] " << msg << std::endl;
}

void AddonLogger::warn(const std::string& prefix, const std::string& msg) {
    std::cout << "[RaveX-Addon] [" << prefix << "] [WARN] " << msg << std::endl;
}

void AddonLogger::error(const std::string& prefix, const std::string& msg) {
    std::cerr << "[RaveX-Addon] [" << prefix << "] [ERROR] " << msg << std::endl;
}

}
}

File: RaveX-ravex146/src/main/java/ravex/addon/AddonLogger.java
package ravex.addon;

public class AddonLogger {
    private final String prefix;

    public AddonLogger(String name) {
        this.prefix = "[" + name + "] ";
    }

    public void info(String msg) {
        ravex.RaveX.LOGGER.info(prefix + msg);
    }

    public void warn(String msg) {
        ravex.RaveX.LOGGER.warn(prefix + msg);
    }

    public void error(String msg) {
        ravex.RaveX.LOGGER.error(prefix + msg);
    }
}
